ksp_plugin/part.cpp

```diff
@@ -82,16 +82,32 @@ Part::degrees_of_freedom() const {
   return degrees_of_freedom_;
 }
 
-DiscreteTrajectory<Barycentric>& Part::tail() {
+DiscreteTrajectory<Barycentric>::Iterator Part::history_begin() {
+  // Make sure that we skip the point of the prehistory.
+  auto it = history_->Fork();
+  return ++it;
+}
+
+DiscreteTrajectory<Barycentric>::Iterator Part::history_end() {
+  return history_->End();
+}
+
+DiscreteTrajectory<Barycentric>::Iterator Part::psychohistory_begin() {
   if (psychohistory_ == nullptr) {
     psychohistory_ = history_->NewForkAtLast();
   }
-  return *psychohistory_;
+  // Make sure that we skip the fork, which may be the point of the prehistory.
+  auto it = psychohistory_->Fork();
+  return ++it;
```

Comment:
This seems like a very confusing API: the points in [psychohistory_begin, psychohistory_end[ are missing a bit of the psychohistory (the range between the fork and psychohistory_begin) when the psychohistory is forked in the history. When treating the trajectiories as continuous, this will cause issues.

If this is really the intended behaviour, rename it to something other than begin (first_ahistorical_iterator?).